app/models/contribution.rb

```diff
@@ -51,4 +59,12 @@ def update_reviewer(reviewer_id)
   scope :active_engineers, -> { joins(:user).merge(User.engineer.active) }
   scope :valid_pull_requests, -> { where.not(state: :refused) }
   scope :without_pr_state, ->(state) { where.not(pr_state: state) }
+
+  private
+
+  def rejected_reason_presence
+    if (approved? || received?) && !rejected_reason.blank?
+      errors.add(:rejected_reason, :must_be_blank, state: Contribution.human_attribute_name("state/#{state}"))
```

Comment:
the method is rejected_reason_presence

and the error key is must_be_blank?

also there's the condition `!rejected_reason.blank?`

Reply:
I see your point. 

This method is used in the `edit` contribution and serves to ensure that `rejected_reason` will only be present in the `refused` state. Basically, I'm verifying if `rejected_reason_presence` presence in the states `received || approved`. If so, raise an error informing that `rejected_reason` must be blank in the approved and received states.

Do you have any suggestions?